UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.CleanUpJob.UnitTests/Services/ExchangeSetCleanUpServiceTest.cs

```diff
@@ -0,0 +1,78 @@
+
+        [SetUp]
+        public void Setup()
+        {
+            fakeAzureFileSystemHelper = A.Fake<IAzureFileSystemHelper>();
+            fakeStorageConfig = Options.Create(new EssFulfilmentStorageConfiguration()
+                                         { StorageContainerName ="Test"  });
+            fakeScsStorageService = A.Fake<ISalesCatalogueStorageService>();
+            fakeConfiguration = A.Fake<IConfiguration>();
+            fakeLogger = A.Fake<ILogger<ExchangeSetCleanUpService>>();
+            fakeCleanUpConfig = Options.Create(new CleanUpConfiguration()
+            { NumberOfDays = 1 });
+
+            exchangeSetCleanUpService = new ExchangeSetCleanUpService(fakeAzureFileSystemHelper, fakeStorageConfig, fakeScsStorageService, fakeConfiguration, fakeLogger, fakeCleanUpConfig);
+        }
+
+        [Test]
+        public void WhenScsStorageAccountAccessKeyValueNotFound_ThenReturnKeyNotFoundException()
+        {
+            A.CallTo(() => fakeScsStorageService.GetStorageAccountConnectionString(A<string>.Ignored, A<string>.Ignored))
+              .Throws(new KeyNotFoundException("Storage account accesskey not found"));
+
+            Assert.ThrowsAsync(Is.TypeOf<KeyNotFoundException>()
+                   .And.Message.EqualTo("Storage account accesskey not found")
+                    , async delegate { await exchangeSetCleanUpService.DeleteHistoricFoldersAndFiles(); });
+        }
+
+        [Test]
+        public async Task WhenHistoricFoldersAndFilesNotFound_ThenReturnFalseResponse()
+        {
+            A.CallTo(() => fakeScsStorageService.GetStorageAccountConnectionString(A<string>.Ignored, A<string>.Ignored)).Returns(fakeStorageAccountConnectionString);
+            A.CallTo(() => fakeAzureFileSystemHelper.DeleteDirectoryAsync(fakeCleanUpConfig.Value.NumberOfDays, fakeStorageAccountConnectionString, fakeStorageConfig.Value.StorageContainerName, fakeFilePath)).Returns(false);
+
+            var response = await exchangeSetCleanUpService.DeleteHistoricFoldersAndFiles();
+
+            Assert.AreEqual(false, response);
+        }
+
+        [Test]
+        public async Task WhenHistoricFoldersAndFilesFound_ThenReturnTrueResponse()
+        {
+            fakeConfiguration["HOME"] = fakeFilePath;
+
+            A.CallTo(() => fakeScsStorageService.GetStorageAccountConnectionString(A<string>.Ignored, A<string>.Ignored)).Returns(fakeStorageAccountConnectionString);
+            A.CallTo(() => fakeAzureFileSystemHelper.DeleteDirectoryAsync(fakeCleanUpConfig.Value.NumberOfDays, fakeStorageAccountConnectionString, fakeStorageConfig.Value.StorageContainerName, fakeFilePath)).Returns(true);
+
+            var response = await exchangeSetCleanUpService.DeleteHistoricFoldersAndFiles();
+
+            Assert.AreEqual(true, response);
+        }
```

Comment:
I think the two tests above re actually testing that the call azureDeleteFileSystemHelper.DeleteDirectoryAsync takes place (which is fine).

However the result of the call should ideally be tested on the azureDeleteFileSystemHelper (see earlier comments)

Reply:
Resolved, I have made required changes for same.